posthog/schema_migrations/base.py

```diff
@@ -0,0 +1,31 @@
+from abc import ABC, abstractmethod
+
+from posthog.schema import NodeKind
+
+
+class SchemaMigration(ABC):
+    """
+    Base class for schema migrations. Bumps the schema of one or more query types by one version.
+
+    Subclasses must define:
+    * `targets`: A mapping of node kinds to the version before running the migration.
+    * `transform`: A method that takes a dict query and returns a new dict query with the updated schema version.
+    """
+
+    targets: dict[NodeKind, int] = {}
+
+    def __call__(self, query: dict) -> dict:
+        """Apply if version matches, otherwise return untouched."""
+        if not self.should_run(query):
+            return query
+        query = self.transform(query)
+        query["v"] = self.targets[query["kind"]] + 1  # bump version
```

Comment:
I'd prefer to just call this `version` as then doesn't require reading docstring to confirm if it's indeed the version :)

Reply:
Done!